test/client_handles_bad_connections_test.dart

```diff
@@ -47,54 +49,57 @@ class FixedConnectionClientChannel extends ClientChannelBase {
+    ),
+  ));
 
-    expect(await testClient.stream(1).toList(), [1, 2, 3]);
-    await Future.delayed(Duration(milliseconds: 200));
-    expect(await testClient.stream(1).toList(), [1, 2, 3]);
-    expect(
-        channel.states.where((x) => x == grpc.ConnectionState.ready).length, 2);
-    server.shutdown();
-  });
+  final testClient = TestClient(channel);
+  expect(await testClient.stream(1).toList(), [1, 2, 3]);
+  await Future.delayed(Duration(milliseconds: 200));
+  expect(await testClient.stream(1).toList(), [1, 2, 3]);
+  expect(
+      channel.states.where((x) => x == grpc.ConnectionState.ready).length, 2);
+  server.shutdown();
+}
 
-  test('client reconnects when stream limit is used', () async {
-    final grpc.Server server = grpc.Server([TestService()]);
-    await server.serve(
-        address: 'localhost',
-        port: 0,
-        http2ServerSettings: ServerSettings(concurrentStreamLimit: 2));
+void testCase02(InternetAddress address) async {
+  final grpc.Server server = grpc.Server([TestService()]);
+  await server.serve(
+      address: address,
+      port: 0,
+      http2ServerSettings: ServerSettings(concurrentStreamLimit: 2));
 
-    final channel = FixedConnectionClientChannel(Http2ClientConnection(
-        'localhost',
-        server.port,
-        grpc.ChannelOptions(credentials: grpc.ChannelCredentials.insecure())));
-    final states = <grpc.ConnectionState>[];
-    channel.clientConnection.onStateChanged =
-        (Http2ClientConnection connection) => states.add(connection.state);
-    final testClient = TestClient(channel);
+  final channel = FixedConnectionClientChannel(Http2ClientConnection(
+      address,
+      server.port,
+      grpc.ChannelOptions(credentials: grpc.ChannelCredentials.insecure())));
+  final states = <grpc.ConnectionState>[];
+  channel.clientConnection.onStateChanged =
+      (Http2ClientConnection connection) => states.add(connection.state);
+  final testClient = TestClient(channel);
 
-    await Future.wait(<Future>[
-      expectLater(testClient.stream(1).toList(), completion([1, 2, 3])),
-      expectLater(testClient.stream(1).toList(), completion([1, 2, 3])),
-      expectLater(testClient.stream(1).toList(), completion([1, 2, 3])),
-      expectLater(testClient.stream(1).toList(), completion([1, 2, 3])),
-    ]);
-    expect(states.where((x) => x == grpc.ConnectionState.ready).length, 2);
-    server.shutdown();
-  });
+  await Future.wait(<Future>[
+    expectLater(testClient.stream(1).toList(), completion([1, 2, 3])),
+    expectLater(testClient.stream(1).toList(), completion([1, 2, 3])),
+    expectLater(testClient.stream(1).toList(), completion([1, 2, 3])),
+    expectLater(testClient.stream(1).toList(), completion([1, 2, 3])),
+  ]);
+  expect(states.where((x) => x == grpc.ConnectionState.ready).length, 2);
+  server.shutdown();
+}
+
+main() async {
+  testTcpAndUds('client reconnects after the connection gets old', 'localhost', testCase01);
```

Comment:
I think we should keep the closures inlined, I think it better for readability even though the first line becomes a bit hard to read, but then you immediately know what the test is testing from the description string.  

```
testTcpAndUds('client reconnects after the connection gets old', (address) async {
  // ...
});
```